src/ocean/text/convert/Integer.d

```diff
@@ -30,6 +30,133 @@ import ocean.transition;
 
 import ocean.core.Traits;
 
+import ocean.core.array.Search : find;
+
+import ocean.util.ReusableException;
+
+import ocean.math.Math;
+
+
+/*******************************************************************************
+
+    Parses a floating point number represented as a string directly to an
+    integer value.
+
+    To represent the fractional part we multiply the value by the requested
+    amount of decimal points and add it up. For example:
+
+    "1.123" -> 1123  (decimal_points = 3)
+    "0.01"  ->   10  (decimal_points = 3)
+
+    Any characters longer than the requested amount of decimal points will be
+    cut off:
+
+    "1.2345"  ->  123 (decimal_points = 2)
+    "10.2030" -> 1020 (decimal_points = 2)
+
+    Params:
+        decimal_points = amount of decimal points to consider
+        T              = type of the integer
+        value          = floating point number string to parse
+        exception      = optional, a reusable exception instance to throw
+
+    Returns:
+        the resulting integer number
+
+*******************************************************************************/
+
+public T floatStringToInt ( size_t decimal_points, T = ulong )
```

Comment:
Well, I figured it is a compile time optimization because that way the `pow()` line is also done at compile time and only the multiplication with the price needs to be done at runtime